jax/_src/linear_util.py

```diff
@@ -349,9 +349,9 @@ def memoized_fun(fun: WrappedFun, *args):
       ans, stores = result
       fun.populate_stores(stores)
     else:
-      ans = call(fun, *args)
       if explain and config.explain_cache_misses.value:
-        explain(fun.f, cache is new_cache, cache, key, ans)
+        explain(fun.f, cache is new_cache, cache, key)
+      ans = call(fun, *args)
```

Comment:
Shouldn't explain run after the call? Maybe keep the order but don't pass in result?

Reply:
I thought it makes sense to explain before, because the miss happens before the call. Wdyt?

Reply:
What if it's the first call? You don't wanna explain before first call?

Reply:
Okay, reverted that bit.